unit_tests/src/test_environment_variables.cpp

```diff
@@ -21,39 +22,54 @@ TEST_CASE("environment variables")
         // Assert the working case:
         // Given a large enough buffer, the environment variable will be coppied into the buffer
         constexpr std::size_t LARGE_BUFFER_SIZE = 100;
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors)
         char large_buffer[LARGE_BUFFER_SIZE]{};
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(0 == capture_string_property(env_var_key.c_str(), large_buffer, LARGE_BUFFER_SIZE));
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(strnlen(large_buffer, LARGE_BUFFER_SIZE) == env_var_val.size());
 
         // Assert it fails with small buffer:
         // If the buffer is too small, capture_string_property should
         // return a SYS_GET_ENV_ERR and the buffer should remain empty
         constexpr std::size_t SMALL_BUFFER_SIZE = 10;
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors)
         char small_buffer[SMALL_BUFFER_SIZE]{};
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(SMALL_BUFFER_SIZE < env_var_val.size());
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(SYS_GETENV_ERR == capture_string_property(env_var_key.c_str(), small_buffer, SMALL_BUFFER_SIZE));
 
         // Assert it fails with small buffer size:
         // If the given buffer size is too small, capture_string_property should
         // return a SYS_GET_ENV_ERR and the buffer should remain empty
         // even if the buffer can actually hold more
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors)
         char mismatched_buffer[LARGE_BUFFER_SIZE]{};
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(SYS_GETENV_ERR == capture_string_property(env_var_key.c_str(), mismatched_buffer, SMALL_BUFFER_SIZE));
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
         REQUIRE(strnlen(mismatched_buffer, SMALL_BUFFER_SIZE) == 0);
 
         // Assert it fails with zero buffer size:
         // If the given buffer size is zero, capture_string_property should
         // return a SYS_GET_ENV_ERR and the buffer should not be touched
         constexpr std::size_t ZERO_BUFFER_SIZE = 0;
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,-warnings-as-errors)
         char zero_buffer[SMALL_BUFFER_SIZE]{};
```

Comment:
I don't think this is valid C++.

We need to investigate whether zero-length arrays are acceptable.

Reply:
I guess I wasn't really clear what I was doing there.

I was testing if you provided a size of zero, would the function throw an error?

the "`zero_buffer`" actually has a size of 10 as it is instantiated with the `SMALL_BUFFER_SIZE`.

Reply:
I misread the code. I somehow thought the buffer was initialized with a size of zero.

I'll look over that one again and leave a comment on whether anything needs to be changed.

Reply:
I think we're good here. Please mark as resolved.